core/blockchain_test.go

```diff
@@ -2241,3 +2241,48 @@ func BenchmarkBlockChain_1x1000Executions(b *testing.B) {
 	}
 	benchmarkLargeNumberOfValueToNonexisting(b, numTxs, numBlocks, recipientFn, dataFn)
 }
+
+// Tests that importing a some old blocks, where the first one is before the
```

Comment:
where the last one is before the pruning point?

Reply:
Additionally, I think the normal case is:

* Downloader rollbacks a batch of inserted blocks and exits
* Downloader starts to sync again
* The blocks fetched are all known blocks

Maybe it's nice to describe this case also

Reply:
> where the last one is before the pruning point?

Yes, all blocks are pre-pruning